Automattic/wp-calypso
# Inline Comments - move isExpanded component state to calypso state

Related to # pe7F0s-17X-p2

## Proposed Changes

* Moves isExpanded state for inline comments out of component state and into calypso redux.
* The new `inlineExpansion` section of the comments state tracks expansion state for inline comments by streamKey, siteId, and postid. Thus expansion state of a posts inline comments should persist on the same stream in the same session, but not across different streams.
* This means that when going from a Stream -> full post -> ' < back' to stream, all expanded comments should still be expanded and the scroll position (and visible posts/comments) will be the same returning as it was when leaving.

As a result, when navigating between streams our scroll position (and which comments are expanded) remain the same:
![between-feeds](https://github.com/Automattic/wp-calypso/assets/28742426/eb80eb3e-863f-4e27-81dd-08d68cea1d0b)

Similarly, when going to view the full post and coming back the scroll position and which comments are expanded also remains the same:
![full-site-and-back](https://github.com/Automattic/wp-calypso/assets/28742426/e342fdfc-d034-4f2d-b443-ae9634713cf0)


## Testing Instructions



* Run this branch of calypso.
* go to a stream with comments to expand (Discover is good for this).
* Expand some comment sections as you scroll down.
* Test clicking a card and viewing a full post. Note what you are looking at on the steam when you leave to view the full post.
* click the `< back` button in the top left when done.  Verify you are at the same location on the stream as when you left.
* smoke test other features around inline comments and full-post comments.

## Pre-merge Checklist
